cmd/setup.go

```diff
@@ -951,11 +951,50 @@ func configureDevices(conf globalconfig.All) error {
 	return joinErrors(errs...)
 }
 
+func migrateModbusProxy() error {
+	var arr []map[string]any
+	err := settings.Yaml(keys.ModbusProxy, new([]map[string]any), &arr)
+	if err != nil {
+		return err
+	}
+
+	for _, obj := range arr {
+		settings := make(map[string]any)
+		for key, value := range obj {
+			if key != "port" && key != "readonly" {
```

Comment:
This manual mapping should not be necessary. YAML/JSON handling should do the correct thing based on the annotations.